xcube/api/gen/gen.py

```diff
@@ -254,10 +256,36 @@ def step8(dataset):
         if append_mode and os.path.exists(output_path):
             # noinspection PyShadowingNames
             def step9(dataset):
-                output_writer.append(dataset, output_path, **output_writer_params)
+                global _APPEND_DS_TO_DC
+                if output_path.endswith('.nc'):
+                    output_writer.append(dataset, output_path, **output_writer_params)
+                elif output_path.endswith('.zarr'):
+                    _APPEND_DS_TO_DC = check_append_or_insert(time_range, output_path)
+                    if _APPEND_DS_TO_DC is True:
+                        output_writer.append(dataset, output_path, **output_writer_params)
+                    elif _APPEND_DS_TO_DC is False:
+                        tempdir_for_merging = os.path.join(os.path.dirname(__file__), '..', '..', '..',
+                                                           'tempdir_for_merging')
+                        try:
+                            os.mkdir(tempdir_for_merging)
+                        except OSError:
+                            monitor("Creation of the directory %s failed" % tempdir_for_merging)
+                        else:
+                            monitor("Successfully created the directory %s for inserting "
+                                    "time stamp " % tempdir_for_merging)
+                        temp_output_path = os.path.join(tempdir_for_merging,
+                                                        input_file.replace("/", "_")[:-3] + '.zarr')
+                        output_writer.write(dataset, temp_output_path, **output_writer_params)
+                        merged_data_set = merge_single_zarr_into_destination_zarr(temp_output_path, output_path)
+                        rimraf(tempdir_for_merging)
+                        if not merged_data_set:
+                            monitor('Time Stamp of input data set is already existing in data cube: skipping...')
+                            return False
+
                 return dataset
 
-            steps.append((step9, f'appending to {output_path}'))
+            steps.append((step9, f'appending or merging to {output_path}'))
```

Comment:
```suggestion
            steps.append((step9, f'appending to or inserting into {output_path}'))
```

or use `_APPEND_DS_TO_DC` (as local variable of course) to decide which text to print.